### PR TITLE
Compilation pipeline

### PR DESCRIPTION
**Pull request by gwwatkin**
Take the compilation pipeline out of the view and put it into it's own file.

**Update.** Slipped in a new feature: shows the compilation pipleine on the click of a button (screenshot below).

### CODE REVIEW

**Comment by gwwatkin** [comment]
@alexnguyenn

File: lattice_surgery_compilation_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import sparse_lattice_to_array
+from visual_array_cell import *
+
+
+GUISlice = List[List[Optional[VisualArrayCell]]] # 2D array of cells
+
+__all__ = ['compile_file','VisualArrayCell','GUISlice']
+
+def compile_file(circuit_file_name : str ,
+                 apply_litinski_transform:bool=True) -> List[GUISlice]:
```
Comment (gwwatkin) [inline lattice_surgery_compilation_pipeline.py:14]: @Keelando We need to add a checkbox in the ui to enable this

**Comment by gwwatkin** [comment]
New feature: show the compilation pipleine on the click of a button:
Screenshot_20210322_035628

**Comment by alexnguyenn** [comment]
The new feature is really nice. My suggestion is to add also the diagram of the circuit after transformation is applied (via `circuit.apply_transformation`).

**Comment by gwwatkin** [comment]
> New feature: show the compilation pipleine on the click of a button:
Created #59 to refine this view

**Review by Keelando — APPROVED** [review]
I will probably have some merge conflicts with my current work as I also cleaned up views.py after the last refactor, but will address those as they come.

**Pull request merged**